lib/Doctrine/DBAL/Platforms/AbstractPlatform.php

```diff
@@ -3578,4 +3578,22 @@ public function getStringLiteralQuoteCharacter()
     {
         return "'";
     }
+
+    /**
+     * Escapes metacharacters in a string intended to be used with a LIKE
+     * operator.
+     */
+    public function escapeStringForLike(string $untrustedString): string
```

Comment:
This is not about trust. The input is an arbitrary string with no meaning while the output is a LIKE expression.

Reply:
It could indeed come from a trusted source and still need to be escaped because you genuinely want to search for that 👍